firebase-firestore/src/androidTest/java/com/google/firebase/firestore/CountTest.java

```diff
@@ -0,0 +1,249 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import static com.google.firebase.firestore.testutil.IntegrationTestUtil.testCollection;
+import static com.google.firebase.firestore.testutil.IntegrationTestUtil.testCollectionWithDocs;
+import static com.google.firebase.firestore.testutil.IntegrationTestUtil.testFirestore;
+import static com.google.firebase.firestore.testutil.IntegrationTestUtil.waitFor;
+import static com.google.firebase.firestore.testutil.IntegrationTestUtil.waitForException;
+import static com.google.firebase.firestore.testutil.TestUtil.map;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.google.firebase.firestore.testutil.IntegrationTestUtil;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class CountTest {
+
+  @After
+  public void tearDown() {
+    IntegrationTestUtil.tearDown();
+  }
+
+  @Test
+  public void testCountQueryEquals() {
+    CollectionReference coll1 = testCollection("foo");
+    CollectionReference coll1_same = coll1.firestore.collection(coll1.getPath());
+    AggregateQuery query1 = coll1.count();
+    AggregateQuery query1_same = coll1_same.count();
+    AggregateQuery query2 =
+        coll1.document("bar").collection("baz").whereEqualTo("a", 1).limit(100).count();
+    AggregateQuery query2_same =
+        coll1.document("bar").collection("baz").whereEqualTo("a", 1).limit(100).count();
+    AggregateQuery query3 =
+        coll1.document("bar").collection("baz").whereEqualTo("b", 1).orderBy("c").count();
+    AggregateQuery query3_same =
+        coll1.document("bar").collection("baz").whereEqualTo("b", 1).orderBy("c").count();
+
+    assertEquals(query1, query1_same);
+    assertEquals(query2, query2_same);
+    assertEquals(query3, query3_same);
+
+    assertEquals(query1.hashCode(), query1_same.hashCode());
+    assertEquals(query2.hashCode(), query2_same.hashCode());
+    assertEquals(query3.hashCode(), query3_same.hashCode());
+
+    assertNotEquals(null, query1);
```

Comment:
Add a test for calling AggregateQuery.equals() with a non-null, non-AggregateQuery object (e.g. a String).